### PR TITLE
JBPM-5409: Stunner - Bug fixing and UX improvements (M1).

### PR DESCRIPTION
**Pull request by romartin**
Hello,
This PR includes different bug fixings, ux improvments and disables some features as well, as for the M1 scope.
See [this ticket](https://issues.jboss.org/browse/JBPM-5409) for the details. 
Thanks!

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDiagramMarshallerTest.java [hunk 1]
```diff
@@ -456,7 +456,7 @@ public void testUnmarshallSimulationProperties() throws Exception {
         assertEquals( Double.valueOf(321),  simulationSet.getMean().getValue() );
     }
 
-    @Test
+    // TODO: Disabled @Test - As intermediate events has been disabled for M1
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDiagramMarshallerTest.java]: ```@Ignore``` is probably better.

**Comment by manstis** [comment]
Assigned to @jlindop as he wants to talk to @romartin about https://github.com/droolsjbpm/kie-wb-distributions/pull/415 before this is merged. @jlindop Please let me know when this can be merged.

**Comment by jlindop** [comment]
@manstis   Will you merge this PR and https://github.com/droolsjbpm/kie-wb-distributions/pull/415 . There is a minor issue but Roger & I can talk about it later - it's very minor compared to the overall improvements in this PR.

**Comment by jlindop** [comment]
+1

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/wires/WiresUtils.java [hunk 2]
```diff
@@ -16,14 +16,22 @@
...
 import com.ait.lienzo.client.core.shape.wires.*;
+import com.ait.lienzo.client.core.types.Point2D;
 import org.kie.workbench.common.stunner.core.client.canvas.AbstractCanvasHandler;
 import org.kie.workbench.common.stunner.core.client.shape.view.ShapeView;
 import org.kie.workbench.common.stunner.core.graph.Edge;
 import org.kie.workbench.common.stunner.core.graph.Node;
 
 public final class WiresUtils {
 
+    public static double[] getAbsolute( final IDrawable<?> shape ) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/wires/WiresUtils.java]: The problem with Point2D is that it's a lienzo class, and this module shoult not depend on it, that's why I'm using double[] at some points on the code. Bette creatting our Point2D in Stunner you think?

Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/wires/WiresUtils.java]: @romartin Erm... the class already imports and uses ```Point2D```...

Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/wires/WiresUtils.java]: ooops.... emm hehe... will check... :)

Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/wires/WiresUtils.java]: ah yes, this is lienzo-extensions sorry, not core... agree!!! good point :)

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/src/main/java/org/kie/workbench/common/stunner/core/definition/shape/GlyphShapeDef.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public abstract String getGlyphDefinitionId( String id );
+
+    @Override
+    public Class<?> getType() {
+        return GlyphShapeDef.class;
+    }
+
+    @Override
+    public String getGlyphDescription( final W element ) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/src/main/java/org/kie/workbench/common/stunner/core/definition/shape/GlyphShapeDef.java]: hehe yes this is not good.. absolutely agree

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/drag/DragControlImpl.java [hunk 4]
```diff
@@ -81,9 +115,32 @@ public void end( final DragEvent event ) {
...
 
-    protected double[] getContainerXY( final Shape shape ) {
-        return new double[]{ shape.getShapeView().getShapeX(),
-                shape.getShapeView().getShapeY() };
+    /**
+     * Setting dragBounds for the shape doesn't work on lienzo side, so
+     * ensure drag does not exceed the canvas bounds.
+     * @param shapeView The shape view instance being drag.
+     */
+    private void ensureDragConstrains( final ShapeView<?> shapeView,
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/drag/DragControlImpl.java]: really!?!?!?  But this is a built-in feature in lienzo???? In fact we need just what you say, so how can I achieve that?
PS: For now was only trying to not produce bugs with this, as you could drag shapes outside the canvas and they can never be used again.. so agree, this have to change!!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDiagramMarshallerTest.java (hunk 1 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/BPMNDiagramMarshallerTest.java]: ops sure!

**Pull request merged**